File: tests/14_loadib_1.py
#!/usr/bin/env python3

from interp import *

def t():
    c = [
        0,
        0x400,
        0xfedcba9876543210,
    ]
    i = [
        cpu_encode_op_li_i64(0,0x10),
        cpu_encode_op_loadib_i64(1,0,1),
        cpu_encode_op_break(0),
    ]
    cpu_test("loadib_1", c, i)

t()

File: tests/14_loadib_1.c
#include "glyph.h"

void t()
{
    u64 c[] = {
        0x408,
        0xfedcba9876543210ull,
    };
    u16 i[] = {
        cpu_encode_op_li_i64(0,0),
        cpu_encode_op_loadib_i64(1,0,0),
        cpu_encode_op_break(0)
    };
    cpu_test("loadib_1", c, i);
}

int main(int argc, char **argv)
{
    t();
    return 0;
}
